File: ant-sdk/antd-dart/example/03_chunks.dart
import 'dart:convert';
import 'dart:typed_data';

import 'package:antd/antd.dart';

/// Demonstrates storing and retrieving raw chunks.
void main() async {
  final client = AntdClient();

  try {
    // Store a chunk
    final data = Uint8List.fromList(utf8.encode('raw chunk data'));
    final result = await client.chunkPut(data);
    print('Chunk stored at ${result.address} (cost: ${result.cost} atto)');

    // Retrieve the chunk
    final retrieved = await client.chunkGet(result.address);
    print('Retrieved: ${utf8.decode(retrieved)}');
  } on AntdError catch (e) {
    print('Error: $e');
  } finally {
    client.close();
  }
}

File: ant-sdk/antd-elixir/examples/03_chunks.exs
Mix.install([
  {:antd, path: ".."}
])

# Store and retrieve raw chunks
client = Antd.Client.new()

# Store a chunk
{:ok, result} = Antd.Client.chunk_put(client, "raw chunk data")
IO.puts("Chunk stored at: #{result.address}")
IO.puts("Cost: #{result.cost} atto")

# Retrieve the chunk
{:ok, data} = Antd.Client.chunk_get(client, result.address)
IO.puts("Retrieved chunk: #{data}")
